Add event voting

cheshire137 opened the pull request:
This partially addresses #7 by allowing users to vote on events. You can only vote on events you can't edit, meaning you can't vote on your own event. This uses the [acts_as_votable gem](https://github.com/ryanto/acts_as_votable).

This branch only gives the ability to vote up or down on events. It doesn't provide an index view of events sorted by votes. I wanted to keep the PR diff small, so I think a) voting on records other than events and b) sorting records based on votes could come in separate branches.

I felt upvoting and downvoting looked better with the thumbs up and thumbs down Glyphicons, so I changed the "Flag as Spam" glyphicon to be the circle with a slash in it, instead of the thumbs down.

I noticed when I ran `rake spec`, indication_spec.rb was erroring out because it required a nonexistent rails_helper file. So I updated it to require the spec_helper that other tests require.

<img width="574" alt="google_fundraiser___startup_wichita" src="https://user-images.githubusercontent.com/82317/31580219-bdd9d738-b10e-11e7-848b-5f05a469cff7.png">

cheshire137 committed:
Add acts_as_votable gem

cheshire137 committed:
Add votes table

cheshire137 committed:
Make Event votable

cheshire137 committed:
Add VotesController

cheshire137 committed:
Use ban circle icon for spam flagging

cheshire137 committed:
Flesh out vote create endpoint

cheshire137 committed:
Add upvote/downvote buttons to event show

cheshire137 committed:
Add helper methods #upvoted?, #downvoted?

cheshire137 committed:
Style like/dislike based on vote status

cheshire137 committed:
Add ability to undo vote

cheshire137 committed:
Hook up JavaScript to update forms on vote

cheshire137 committed:
Upvote incorrect require in model test

cheshire137 committed:
Use built-in voted_up_on?, voted_down_on?

cheshire137 committed:
Add some votes controller tests

kentonh commented:
Thanks so much @cheshire137!
One thing. Can you remove downvotes? I just want to have upvotes.

cheshire137 commented:
Yep, that's easy enough!

cheshire137 committed:
Remove ability to downvote, rename upvote to 'recommend'

In response to https://github.com/StartupWichita/startupwichita.com/pull/127#issuecomment-336680109

cheshire137 committed:
Handle bad votable ID without exception

cheshire137 committed:
Test 404 responses when given bad params

cheshire137 commented:
This now only has upvoting. I renamed it to "recommend" because it felt odd to have just a "like" button by itself.

When the user has upvoted:

<img width="569" alt="google_fundraiser___startup_wichita" src="https://user-images.githubusercontent.com/82317/31580986-7c69d936-b124-11e7-8251-fa77d529eae2.png">

When the user has not upvoted:

<img width="568" alt="google_fundraiser___startup_wichita" src="https://user-images.githubusercontent.com/82317/31580989-8cabbe9a-b124-11e7-9cfa-a7035b01ee4e.png">

kentonh committed:
Merge branch 'master' into add-event-voting
